src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -370,87 +383,167 @@ function addHeadAndTrailingLogToBreach({ result, sensor, database }) {
+    .max('timestamp');
+  const maxTimestampForDeletion = new Date().getTime() - SENSOR_LOG_NO_DELETION_INTERVAL;
+  // Starting timestamp is either from the first fullAggregate timestamp, or beginning of time
+  const aggregationIntervalStart = (mostRecentFullAggregateTimestamp || -Infinity) + 1;
+  // Ending timestamp is either from the most recent preAggregate or from now, less the
+  // no full aggregate period (32 hours)
+  const aggregationIntervalEnd =
+    (mostRecentPreAggregateTimestamp || new Date().getTime()) - NO_FULL_AGGREGATE_PERIOD;
+  // Query for all sensorLogs which are preAggregates, between the above two
+  // timestamps
+  const sortedSensorLogs = sensorLogs
+    .filtered(
+      'aggregation == $0  && timestamp > $1 && timestamp < $2',
+      SENSOR_LOG_PRE_AGGREGATE_TYPE,
+      aggregationIntervalStart,
+      aggregationIntervalEnd
+    )
+    .sorted('timestamp');
+  // Will hold the timestamp of a sequential group of preAggregate logs
+  // which are to-be aggregated into a fullAggregate.
+  let logGroupStartTimestamp = Infinity;
+  // sensorLogGroups is a 2D array for all groups of to-be-aggregated sensor logs.
+  // sensorLogGroup is A collection of sensor logs which will be aggregated into a
+  // full aggregated sensorLog - 1D arrays housed within the 2D array.
+  const sensorLogGroups = [];
+  let sensorLogGroup = [];
+  // Find sets of sensorLogs within the full aggregation interval. Delimited by
+  // a sensorLog which is not within the same interval.
+  sortedSensorLogs.forEach(sensorLog => {
+    const { timestamp } = sensorLog;
+    // Premature return if this sensorlog is malformed
+    if (!timestamp) return;
+    // Check if this sensorLog is in the current interval. The first iteration
+    // will never be in the same interval, as one hasn't been created/started.
+    const isInSameInterval = logGroupStartTimestamp - timestamp < FULL_AGGREGATION_INTERVAL;
+    // If so, add it to the current group of logs.
+    if (isInSameInterval) sensorLogGroup.push(sensorLog);
+    // Otherwise, hit a delimiter. If there are already logs in the current
+    // sensor log group, this is an ending delimiter, store the group and reset.
+    else if (sensorLogGroup.length !== 0) sensorLogGroups.push([...sensorLogGroup]);
+    // Resetting for a new group of sensor log. This is called on the first iteration
+    // to initialize the first group. The last group of sensorLogs will not be set aside
+    // for aggregation unless it is also delimited by a sensorLog not within that group.
+    if (!isInSameInterval) {
+      sensorLogGroup = [sensorLog];
+      logGroupStartTimestamp = timestamp;
+    }
   });
-
+  // Create an array of aggregated sensor log objects
+  const aggregatedSensorLogs = sensorLogGroups
+    .map(group => createFullAggregateSensorLogs(group))
+    .flat();
+  const sensorLogsToDelete = sortedSensorLogs.filtered('timestamp < $0', maxTimestampForDeletion);
```

Comment:
this could mean the logs at the end of sortedSensorLogs, that are not aggregated yet (i.e. time between them is less then FULL_AGGREGATION_INTERVAL), would be deleted ?

Reply:
da613bfbbc6d41ab60e2c213d675915ab62295fe